molgenis-genomebrowser/src/main/java/org/molgenis/genomebrowser/GenomeBrowserTrack.java

```diff
@@ -0,0 +1,120 @@
+package org.molgenis.genomebrowser;
+
+import com.google.auto.value.AutoValue;
+
+import javax.annotation.Nullable;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.molgenis.data.meta.model.EntityType;
+import org.molgenis.genomebrowser.meta.GenomeBrowserAttributes;
+import org.molgenis.genomebrowser.meta.GenomeBrowserSettings;
+
+import java.util.stream.Collectors;
+
+@AutoValue
+public abstract class GenomeBrowserTrack
+{
+	public static GenomeBrowserTrack create(String id, String labelAttr, EntityType entity,
+			GenomeBrowserSettings.TrackType trackType, Iterable<GenomeBrowserTrack> molgenisReferenceTracks,
+			GenomeBrowserSettings.MolgenisReferenceMode molgenisReferenceMode,
+			GenomeBrowserAttributes genomeBrowserAttrs, String actions, String attrs, String scoreAttr, String exonKey)
+	{
+		return new AutoValue_GenomeBrowserTrack(id, labelAttr, entity, trackType, molgenisReferenceTracks,
+				molgenisReferenceMode, genomeBrowserAttrs, actions, attrs, scoreAttr, exonKey);
+	}
+
+	public static GenomeBrowserTrack create(GenomeBrowserSettings settings)
+	{
+		return new AutoValue_GenomeBrowserTrack(settings.getIdentifier(), settings.getLabelAttr().getName(),
+				settings.getEntity(), settings.getTrackType(),
+				settings.getMolgenisReferenceTracks().collect(Collectors.toList()), settings.getMolgenisReferenceMode(),
+				settings.getGenomeBrowserAttrs(), settings.getActions(), settings.getAttrs(), settings.getScoreAttr(),
+				settings.getExonKey());
+	}
+
+	@Override
+	public String toString()
```

Comment:
AutoValue toString() generates:
```java
@Override
  public String toString() {
    return "GenomeBrowserTrack{"
        + "id=" + id + ", "
        + "labelAttr=" + labelAttr + ", "
        + "entity=" + entity + ", "
        + "trackType=" + trackType + ", "
        + "molgenisReferenceTracks=" + molgenisReferenceTracks + ", "
        + "molgenisReferenceMode=" + molgenisReferenceMode + ", "
        + "genomeBrowserAttrs=" + genomeBrowserAttrs + ", "
        + "actions=" + actions + ", "
        + "attrs=" + attrs + ", "
        + "scoreAttr=" + scoreAttr + ", "
        + "exonKey=" + exonKey
        + "}";
  }
```
So You do not need to implement toString() yourself